### PR TITLE
Submit empty form values

### PR DESCRIPTION
**Pull request by b-raines**
react-final-form does not submit values for inputs w/ undefined values

https://github.com/marmelab/react-admin/issues/3551

### CODE REVIEW

**Comment by natrim** [comment]
checkForInitialValues should be export from folder to allow custom imports from outside of package to create custom forms

File: packages/ra-ui-materialui/src/form/checkForInitialValues.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import merge from 'lodash/merge';
+
+const checkForInitialValues = (initialValues, newValues) => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/checkForInitialValues.js]: by reading that method name, I don't understand what it's supposed to do or to return. Can you add jsDoc, example usage, or change the name?

File: packages/ra-ui-materialui/src/form/checkForInitialValues.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    // library issue via https://github.com/final-form/react-final-form/issues/130#issuecomment-493447888
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/checkForInitialValues.js]: What does that comment mean?

File: packages/ra-ui-materialui/src/form/checkForInitialValues.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                newValues[key]
+            );
+        } else {
+            acc[key] =
+                typeof newValues[key] === 'undefined' ? '' : newValues[key];
+        }
+        return acc;
+    }, {});
+
+    // need to deep merge to get new child properties
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/checkForInitialValues.js]: I don't understand what you're doing here

File: packages/ra-ui-materialui/src/form/checkForInitialValues.spec.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            age: 30,
+            email: null,
+            job: { title: undefined },
+        };
+        const finalValues = checkForInitialValues(initialValues, values);
+        expect(finalValues).toStrictEqual({
+            firstName: 'John',
+            lastName: '',
+            age: 30,
+            email: null,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/checkForInitialValues.spec.js]: I don't understand the distinction between null and undefined here

Comment (djhi) [reply packages/ra-ui-materialui/src/form/checkForInitialValues.spec.js]: I think `undefined` is when the field has never been set whereas `null` is used explicitly

**Comment by djhi** [comment]
Superseded by #3758 for rebase and review

**Pull request closed without merging**